src/app/settings/views/Configuration/Security/VaultSettings/VaultSettings.tsx

```diff
@@ -0,0 +1,144 @@
+import { useEffect } from "react";
+
+import {
+  Icon,
+  Spinner,
+  CodeSnippet,
+  CodeSnippetBlockAppearance,
+} from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import docsUrls from "app/base/docsUrls";
+import { actions as controllerActions } from "app/store/controller";
+import controllerSelectors from "app/store/controller/selectors";
+import { actions as generalActions } from "app/store/general";
+import { vaultEnabled as vaultEnabledSelectors } from "app/store/general/selectors";
+import type { RootState } from "app/store/root/types";
+
+export enum Labels {
+  Loading = "Loading...",
+  IntegrateWithVault = "Integrate with Vault",
+  VaultEnabled = "Vault enabled",
+  SetupInstructions = "Vault setup instructions",
+  SecretMigrationInsctructions = "Secret migration instructions",
+}
+
+const VaultSettings = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const controllersLoading = useSelector(controllerSelectors.loading);
+  const vaultEnabledLoading = useSelector(vaultEnabledSelectors.loading);
+  const vaultEnabled = useSelector((state: RootState) =>
+    vaultEnabledSelectors.get(state)
+  );
+
+  const { unconfiguredControllers, configuredControllers } = useSelector(
+    (state: RootState) =>
+      controllerSelectors.getVaultConfiguredControllers(state)
+  );
+
+  useEffect(() => {
+    dispatch(controllerActions.fetch());
+    dispatch(generalActions.fetchVaultEnabled());
+  }, [dispatch]);
+
+  if (controllersLoading || vaultEnabledLoading)
+    return <Spinner aria-label={Labels.Loading} text={Labels.Loading} />;
+
+  if (vaultEnabled) {
+    return (
+      <>
+        <p>
+          <Icon name="security-tick" />
+          <span className="u-nudge-right--small">Vault enabled</span>
+        </p>
+        <a href={docsUrls.aboutNativeTLS}>More about Vault integration</a>
+      </>
+    );
+  } else
+    return (
+      <>
+        {unconfiguredControllers.length >= 1 ? (
+          <>
+            {configuredControllers.length >= 1 ? (
+              <p>
+                <Icon name="security-warning" />
+                <span className="u-nudge-right--small">
+                  Incomplete Vault integration, configure{" "}
+                  {unconfiguredControllers.length} other{" "}
+                  {unconfiguredControllers.length > 1
+                    ? "controllers"
+                    : "controller"}{" "}
+                  with Vault to complete this operation.
+                </span>
+              </p>
+            ) : (
+              <p>
+                <Icon name="security" />
+                <span className="u-nudge-right--small">
+                  Integrate with Vault
+                </span>
+              </p>
+            )}
+            <div aria-label={Labels.SetupInstructions}>
```

Comment:
```suggestion
            <section aria-label={Labels.SetupInstructions}>
```

Or even better if the below is possible:
```suggestion
            <section aria-labelledby={id-for-the-heading-containing-"Integrate with Vault"}>
```

Reply:
Absolutely possible, committing it now